### PR TITLE
Removed Wagmi Wrapper 

### PR DESCRIPTION
**Pull request by Adebesin-Cell**
# Removed Wagmi Wrapper 

This PR removes wagmi wrapper to improve performance and utilized better auth system using cookies and verification usin signer.

## Changes
- Refactored ``useFetchWalletBalance``
- Reworked ``isAuthentciatedRoute`` and login page
- Refactored ``useWeb3Token``
- Also changed rpc url as the initial one won't load up.

## Linked issues

Fixes https://github.com/EveripediaNetwork/issues/issues/2221
Fixes https://github.com/EveripediaNetwork/issues/issues/2279

### CODE REVIEW

**Review by Simplypheyie96 — APPROVED** [review]

**Comment by Tolu-Mals** [comment]
I logged in and logged out, but my avatar persisted on the navbar 

https://github.com/EveripediaNetwork/ep-ui/assets/30352484/27a851d8-8291-4d56-a9c1-6da852a406c1

**Comment by Adebesin-Cell** [comment]
> I logged in and logged out, but my avatar persisted on the navbar
> 
>  Screen.Recording.2024-02-13.at.12.24.19.PM.mov

I see, I added a commit, should fix it.

**Review by Softdev1 — APPROVED** [review]

**Review by Royal-lobster — APPROVED** [review]

File: src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx [hunk 1]
```diff
@@ -36,6 +35,7 @@ import networkMap from '@/data/NetworkMap'
 import { ProviderDataType } from '@/types/ProviderDataType'
 import detectEthereumProvider from '@metamask/detect-provider'
```
Comment (Royal-lobster) [inline src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:36]: Do we still need the metamask provider ? cuz now we are using rainbowkit ? I believe it is the one causing the switch network model issue - the chainId from this is returning undefined

**Review by Aliiiu — APPROVED** [review]

File: src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx (hunk 1 above)
Comment (Aliiiu) [reply src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:36]: The chainId is correct, I think the issue is with the rpc_url

Comment (Royal-lobster) [reply src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:36]: I wonder if this is to do with our RPC URL cuz this one is coming from metamask...

CleanShot 2024-02-19 at 15 26 58

 Or you mean the RPC url set in metamask ?
 
CleanShot 2024-02-19 at 15 28 29

Comment (Adebesin-Cell) [reply src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:36]: I'll inspect it.

**Review by Simplypheyie96 — APPROVED** [review]

File: src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx (hunk 1 above)
Comment (Adebesin-Cell) [reply src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:36]: > I wonder if this is to do with our RPC URL cuz this one is coming from metamask...
> 
> CleanShot 2024-02-19 at 15 26 58
> 
> Or you mean the RPC url set in metamask ?
> 
> CleanShot 2024-02-19 at 15 28 29

It does work properly, I added toasts for  feedbacks tho!

**Review by Simplypheyie96 — APPROVED** [review]

File: src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx (hunk 1 above)
Comment (Royal-lobster) [reply src/components/CreateWiki/CreateWikiTopBar/WikiPublish/WikiPublishButton.tsx:36]: Cool, i think we can merge this and sync with editors having issues. tho i still wonder why we need the metamask provider cuz we are now interacting with metamask via rainbow kit now as opposed to before when we were using it directly. probably the detectEthereumProvider can be replaced with a more general function from viem ?

**Review by SidharthK2 — APPROVED** [review]

**Pull request merged**